MetaMask/metamask-extension
# feat: [EXT] Enable STX by default for all users

## **Description**

THIS PR IS CURRENTLY JUST A DRAFT _~Eric Bishard_

This PR enables Smart Transactions (STX) by default for all users who have previously opted out but have no recorded STX in their activity. The following logic is implemented:

1. If the opt-in status (`STX_OPT_IN`) is `null`, it will now default to `true`, and a notification flag will be set to inform users.
2. If the opt-in status is `false`:
   - If the user has no STX activity, the opt-in status will be set to `true`, and a notification flag will be set.
   - If the user has existing STX activity, no changes will be made.
3. If the opt-in status is `true`, no changes are made.

A banner alert will notify users affected by the changes when transitioning from `false` to `true`.

**Target release:** 12.9 (Release Candidate: November 28th)  
**Affected user base:** ~5.7M users who previously opted out of STX but have no STX activity.

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/28568?quickstart=1)

---

## **Related issues**

Fixes: N/A

---

## **Manual testing steps**

1. Set the opt-in status (`STX_OPT_IN`) to `null`.  
   - Verify that the status changes to `true` on migration and that a notification flag is set.
2. Set the opt-in status to `false` with no STX in activity.  
   - Verify that the status changes to `true` on migration and that a notification flag is set.
3. Set the opt-in status to `false` with existing STX activity.  
   - Verify that no changes are made.
4. Set the opt-in status to `true`.  
   - Verify that no changes are made.
5. Trigger app startup with each scenario to ensure the migration works as expected and the notification flag is correctly checked.

---

## **Screenshots/Recordings**

### **Before**



### **After**



---

## **Pre-merge author checklist**

- [x] I've completed the PR template to the best of my ability. (currently a DRAFT)
- [ ] I’ve included tests for the new behavior, covering:
  - [ ] All logic branches:
    - [ ] `null` opt-in status.
    - [ ] `false` opt-in status with no STX activity.
    - [ ] `false` opt-in status with existing STX activity.
    - [ ] `true` opt-in status.
  - [ ] Notification flag setting.

---

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pulled and built the branch, ran the app, and tested the changes described above).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket and includes the necessary testing evidence (e.g., recordings, screenshots, or detailed descriptions).
